src/rpc/mining.cpp

```diff
@@ -179,9 +179,11 @@ static UniValue generatetodescriptor(const JSONRPCRequest& request)
         throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, strprintf("Cannot derive script without private keys"));
     }
 
+    const CTxMemPool& mempool = EnsureMemPool();
```

Comment:
nit: place this either at the start of the function (so it exits early if there is no mempool), or immediately before the call to `generateBlocks()`. Placing it between the code that creates `coinbase_script` and the coinbase_script CHECK seems random.

Same comment for `generatetoaddress()`

Reply:
Might do in a follow-up to not invalidate the three reviews